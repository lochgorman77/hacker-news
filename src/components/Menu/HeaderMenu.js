import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { FormattedMessage } from 'react-intl';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Button from '@material-ui/core/Button';
import { Link as RouterLink } from 'react-router-dom';
import Link from '@material-ui/core/Link';

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
    paddingBottom: 8,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    ...theme.typography.button,
  },
  grow: {
    flexGrow: 1,
  },
}));

export default function HeaderMenu() {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <Button>
            <Link component={RouterLink} to="/">
              <FormattedMessage id={'menu.link.home'} />
            </Link>
          </Button>
          <Button color="secondary">
            <Link component={RouterLink} to="/topstories">
              <FormattedMessage id={'menu.link.top'} />
            </Link>
          </Button>
          <Button>
            <Link component={RouterLink} to="/newstories">
              <FormattedMessage id={'menu.link.new'} />
            </Link>
          </Button>
          <div className={classes.grow} />
          <Button color="inherit">
            <FormattedMessage id={'menu.link.login'} />
          </Button>
        </Toolbar>
      </AppBar>
    </div>
  );
}
