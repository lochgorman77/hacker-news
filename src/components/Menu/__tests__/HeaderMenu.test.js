import React from 'react';
import HeaderMenu from '../HeaderMenu';

const props = {};

describe('<Detail />', () => {
  let shallow, wrapper;

  beforeEach(() => {
    shallow = createShallow();
    wrapper = shallow(<HeaderMenu {...props} />);
  });

  it('renders correctly', () => {
    expect(wrapper.length).toBe(1);
    expect(wrapper.name()).toBe('div');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
