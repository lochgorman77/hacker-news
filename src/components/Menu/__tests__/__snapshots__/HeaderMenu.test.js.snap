// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Detail /> renders correctly 1`] = `
<div
  className="undefined-root"
>
  <MuiAppBar
    position="static"
  >
    <MuiToolbar>
      <MuiButton>
        <MuiLink
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          to="/"
        >
          <FormattedMessage
            id="menu.link.home"
          />
        </MuiLink>
      </MuiButton>
      <MuiButton
        color="secondary"
      >
        <MuiLink
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          to="/topstories"
        >
          <FormattedMessage
            id="menu.link.top"
          />
        </MuiLink>
      </MuiButton>
      <MuiButton>
        <MuiLink
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          to="/newstories"
        >
          <FormattedMessage
            id="menu.link.new"
          />
        </MuiLink>
      </MuiButton>
      <div
        className="undefined-grow"
      />
      <MuiButton
        color="inherit"
      >
        <FormattedMessage
          id="menu.link.login"
        />
      </MuiButton>
    </MuiToolbar>
  </MuiAppBar>
</div>
`;
